src/Init/Tactics.lean

```diff
@@ -461,11 +461,14 @@ syntax config := atomic(" (" &"config") " := " withoutPosition(term) ")"
 /-- The `*` location refers to all hypotheses and the goal. -/
 syntax locationWildcard := " *"
 
+/-- The `⊢` location refers to the current goal. -/
+syntax locationTurnstile := patternIgnore(atomic("|" noWs "-") <|> "⊢")
+
 /--
 A hypothesis location specification consists of 1 or more hypothesis references
 and optionally `⊢` denoting the goal.
 -/
-syntax locationHyp := (ppSpace colGt term:max)+ patternIgnore(ppSpace (atomic("|" noWs "-") <|> "⊢"))?
+syntax locationHyp := (ppSpace colGt (term:max <|> locationTurnstile))+
```

Comment:
Does the docstring show up in hovers? If so, maybe polish the wording a bit while you are at it.

Reply:
It generally doesn't appear in hovers because it's superseded by either the `locationTurnstile` docstring or hover info for local hypotheses. (Hovering over `at` gives the docstring for `location`.) The only time I could get it to appear was when hovering over an invalid identifier in a `locationHyp`. This makes the plurality of the following rephrasing I was considering potentially somewhat confusing, since it appears when hovering over a *particular* (invalid) hypothesis:

> A sequence of locations at which a tactic should operate. These can include local hypotheses and `⊢`, which denotes the goal.

On the other hand, `locationHyp` does represent the full sequence, so phrasing the docstring singularly would make the code itself confusing, which I suspect is more troublesome.

Reply:
I'd say if in doubt, optimize for user visible hovers. Comments for people reading the code can be normal code comments.

Reply:
I've updated the docstring to use a variation on the wording quoted above. I think this is still an improvement clarity-wise, even in the rare cases it appears on hover, and using the plural maintains consistency with hover docstrings of other sequential syntax elements (e.g., tactic blocks).